simulations/vip-502/basemainnet.ts

```diff
@@ -0,0 +1,117 @@
+    redstoneOracle = new ethers.Contract(basemainnet.REDSTONE_ORACLE, CHAINLINK_ORACLE_ABI, provider);
+    resilientOracle = new ethers.Contract(basemainnet.RESILIENT_ORACLE, RESILLIENT_ORACLE_ABI, provider);
+    boundValidator = new ethers.Contract(BOUND_VALIDATOR, BOUND_VALIDATOR_ABI, provider);
+    xvsBridgeAdmin = await ethers.getContractAt(XVS_BRIDGE_ADMIN_ABI, XVS_BRIDGE_ADMIN);
+    xvsBridge = await ethers.getContractAt(XVS_BRIDGE_ABI, XVS_BRIDGE);
+    treasury = await ethers.getContractAt(TREASURY_ABI, basemainnet.VTREASURY);
+
+    await pretendExecutingVip(await vip003());
+    await pretendExecutingVip(await vip004());
+    await pretendExecutingVip(await vip005());
+    await pretendExecutingVip(await vip006());
+    await pretendExecutingVip(await vip007());
+  });
+
+  testForkedNetworkVipCommands("vip502", await vip502());
+
+  describe("Post-VIP behavior", async () => {
+    it(`correct owner `, async () => {
+      expect(await prime.owner()).to.equal(basemainnet.NORMAL_TIMELOCK);
+      expect(await plp.owner()).to.equal(basemainnet.NORMAL_TIMELOCK);
+    });
+
+    it(`correct owner for psr`, async () => {
+      const psr = new ethers.Contract(PSR, PSR_ABI, provider);
+      expect(await psr.owner()).to.equal(basemainnet.NORMAL_TIMELOCK);
+    });
+
+    for (const comptrollerAddress of COMPTROLLERS) {
+      it(`correct owner for ${comptrollerAddress}`, async () => {
+        const c = new ethers.Contract(comptrollerAddress, COMPTROLLER_ABI, provider);
+        expect(await c.owner()).to.equal(basemainnet.NORMAL_TIMELOCK);
+      });
+    }
+
+    for (const vTokenAddress of VTOKENS) {
+      it(`correct owner for ${vTokenAddress}`, async () => {
+        const v = new ethers.Contract(vTokenAddress, VTOKEN_ABI, provider);
+        expect(await v.owner()).to.equal(basemainnet.NORMAL_TIMELOCK);
+      });
+    }
+
+    it("should have the correct pending owner", async () => {
+      expect(await xvsVaultProxy.admin()).to.equal(basemainnet.NORMAL_TIMELOCK);
+      expect(await xvsStore.admin()).to.equal(basemainnet.NORMAL_TIMELOCK);
+    });
+
+    it("XVSBridgeAdmin ownership transferred to Normal Timelock", async () => {
+      expect(await xvsBridgeAdmin.owner()).to.be.equals(basemainnet.NORMAL_TIMELOCK);
+    });
+    it("Normal Timelock should be whitelisted", async () => {
+      expect(await xvsBridge.whitelist(basemainnet.NORMAL_TIMELOCK)).to.be.true;
+    });
+    it("oracles should have correct owner", async () => {
+      expect(await resilientOracle.owner()).equals(basemainnet.NORMAL_TIMELOCK);
+      expect(await chainLinkOracle.owner()).equals(basemainnet.NORMAL_TIMELOCK);
+      expect(await redstoneOracle.owner()).equals(basemainnet.NORMAL_TIMELOCK);
+      expect(await boundValidator.owner()).equals(basemainnet.NORMAL_TIMELOCK);
+    });
+    it("Normal Timelock should be the owner of the Vtreasury", async () => {
+      expect(await treasury.owner()).equals(basemainnet.NORMAL_TIMELOCK);
```

Comment:
I would add an extra check for VTreasury, withdrawing some funds to the normal timelock and checking they are received